lib/fpm/package/deb.rb

```diff
@@ -905,7 +905,7 @@ def mkdir_p(dir)
   def to_s(format=nil)
     # Default format if nil
     # git_1.7.9.3-1_amd64.deb
-    return super("NAME_FULLVERSION_ARCH.TYPE") if format.nil?
+    return super("NAME_FULLVERSION_ARCH.EXTENSION") if format.nil?
```

Comment:
Indeed! Good observation. In hindsight, maybe .to_s primarily for
formatting file names wasn't a great move.

On Monday, April 18, 2016, Paweł Tomulik <notifications@github.com> wrote:

> In lib/fpm/package/deb.rb
> <https://github.com/jordansissel/fpm/pull/1093#discussion_r60172211>:
>
> > @@ -905,7 +905,7 @@ def mkdir_p(dir)
> >    def to_s(format=nil)
> >      # Default format if nil
> >      # git_1.7.9.3-1_amd64.deb
> > -    return super("NAME_FULLVERSION_ARCH.TYPE") if format.nil?
> > +    return super("NAME_FULLVERSION_ARCH.EXTENSION") if format.nil?
>
> Another thought -- to_s is usually meant to represent object in a string
> form for displaying purposes. Here it looks like it has a purpose of
> generating output file name. Is it intentional? Shouldn't these two things
> be separated?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/jordansissel/fpm/pull/1093/files/af92f4976143ff7a2932efcf79d69e4de35ab8a9#r60172211>
>